improver/calibration/rainforest_calibration.py

```diff
@@ -397,6 +400,49 @@ def _make_decreasing(self, probability_data: ndarray) -> ndarray:
         )
         return 0.5 * (upper + lower)
 
+    def _evaluate_probabilities(
+        self,
+        forecast_data: ndarray,
+        input_data: ndarray,
+        output_data: ndarray,
+        preprocessing_fun: Optional[Callable[[ndarray], object]] = None,
+    ):
+        """Evaluate probability that error in forecast exceeds thresholds, setting
+        the result to 1 when `forecast + threshold <= 0`.
+
+        Args:
+            forecast_data:
+                1-d containing data for the variable to be calibrated.
+            input_data:
+                2-d array of data for the feature variables of the model
+            output_data:
+                array to populate with output; will be modified in place
+            preprocessing_fun:
+                function to convert data from ndarray to model input format,
+                only used for treelite predictor
+        """
+
+        if preprocessing_fun:
+            input_dataset = preprocessing_fun(input_data)
+        else:
+            input_dataset = input_data
+
+        for threshold_index, model in enumerate(self.tree_models):
+            threshold = self.error_thresholds[threshold_index]
+            if threshold >= 0:
```

Comment:
The zero here is a bit of a magic number. Essentially what we are trying to exploit here is the fact that when there is a lower bound of the diagnostic in question, there is an associated lower bound on the forecast error: namely error >= - forecast value.

To keep this general, we should think about how we can ensure that the code would work for any parameter, even those which don't have a lower bound. One solution could be to infer the lower bound from ECC bounds, and where none bound exists set the threshold value to be `np.NINF`? Some effort might also be required to appropriately handle units, but this should overcome the magic number issue.

Reply:
I've updated the code to get bounds from `BOUNDS_FOR_ECDF`

Reply:
I think it's best to leave as is for now, and generalise further in a future PR if required.